### PR TITLE
Add support for activity scoped navigation request handlers

### PR DESCRIPTION
**Pull request by deepueg**
This gives more flexibility for applications that follows a complex UI design

### CODE REVIEW

File: android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java [hunk 1]
```diff
@@ -9,6 +9,8 @@
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentManager;
 
+import com.ern.api.impl.navigation.NavigationRouteHandler;
```
Comment (friederbluemle) [inline android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java:12]: Hmm, so this creates a dependency from `.core` to `.navigation`. I'm pretty sure we already have several other such references, and there's already a circular dependency between these two packages.
No action item right now, but we need to remember for "Version 2.0" to rethink the package/dependency structure we're using here. Java package constraints can also be enforced through unit test via tools like [JDepend](https://github.com/clarkware/jdepend). Something we can consider eventually.

File: android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java [hunk 2]
```diff
@@ -91,6 +93,18 @@
...
+    /**
+     * Use this when you don't need each fragment to register it's own navigation request handlers.
+     * When true: {@link com.ern.api.impl.navigation.ReactNavigationViewModel} will be created with activity scope instead of fragment scope.
+     * The activity will then delegate the navigation requests to the corresponding fragments via {@link com.ern.api.impl.navigation.NavigationRouteHandler} implementation.
+     * The {@link ElectrodeBaseActivityDelegate} will take the current instance of the {@link Fragment} that is loaded inside {@link LaunchConfig#getFragmentContainerId()}. Assumption is that this fragment implements {@link NavigationRouteHandler}
+     * Pass {@link RouteHandlerProvider} If you would like to provide your own implementation of providing the fragment instance, valid only if #value is true.
+     */
+    boolean mUseActivityScopedNavigation;
+
+
```
Comment (friederbluemle) [inline android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java:105]: Double blank line here

File: android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java [hunk 3]
```diff
@@ -252,4 +266,26 @@ public void setReplace(boolean replace) {
     public boolean shouldReplace() {
         return mReplace;
     }
+
+    /**
+     * Use this when you don't need each fragment to register it's own navigation request handlers.
+     * When true: {@link com.ern.api.impl.navigation.ReactNavigationViewModel} will be created with activity scope instead of fragment scope.
+     * The activity will then delegate the navigation requests to the corresponding fragments via {@link com.ern.api.impl.navigation.NavigationRouteHandler} implementation.
+     * The {@link ElectrodeBaseActivityDelegate} will take the current instance of the {@link Fragment} that is loaded inside {@link LaunchConfig#getFragmentContainerId()}. Assumption is that this fragment implements {@link NavigationRouteHandler}
+     * Pass {@link RouteHandlerProvider} If you would like to provide your own implementation of providing the fragment instance, valid only if #value is true.
```
Comment (friederbluemle) [inline android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java:275]: No real action here, but a todo for an eventual cleanup: You probably noticed that these five lines are a one-to-one copy+paste from lines 97-101.

File: android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java (hunk 1 above)
Comment (deepueg) [reply android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java:12]: Very good find, this was an oversight. There shouldn't be this dependency.  But I agree, looking at how this is being used we may no longer need a core package as this was initially kept for backward compatibility but no one is actually using it. 

I will fix the dependency now. Thanks.

File: android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java (hunk 3 above)
Comment (deepueg) [reply android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java:275]: removed.

File: android/lib/src/main/java/com/ern/api/impl/navigation/NavigationLaunchConfig.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+     * The activity will then delegate the navigation requests to the corresponding fragments via {@link com.ern.api.impl.navigation.NavigationRouteHandler} implementation.
+     * The {@link ElectrodeBaseActivityDelegate} will take the current instance of the {@link Fragment} that is loaded inside {@link LaunchConfig#getFragmentContainerId()}. Assumption is that this fragment implements {@link NavigationRouteHandler}
+     * Pass {@link RouteHandlerProvider} If you would like to provide your own implementation of providing the fragment instance, valid only if #value is true.
+     */
+    public void useActivityScopeForNavigation(boolean value, @Nullable RouteHandlerProvider<? extends Fragment> routeHandlerProvider) {
+        mUseActivityScopedNavigation = value;
+        mRouteHandlerProvider = routeHandlerProvider;
+    }
+
+    public interface RouteHandlerProvider<T extends Fragment & NavigationRouteHandler> {
```
[the remark this answers is not in the document]
Comment (deepueg) [reply android/lib/src/main/java/com/ern/api/impl/navigation/NavigationLaunchConfig.java:26]: yeah, its pretty neat to have this at compile time.

[the remark this answers is not in the document]
Comment (deepueg) [reply android/lib/src/main/java/com/ern/api/impl/navigation/NavigationLaunchConfig.java:26]: > The imports look a lot cleaner now. Nice! 👍
> 
> I assume you tested all possible scenarios locally? I'm sure you also have the best knowledge how this change will affect the public API and what needs to be done in terms of versioning. I already approved, nothing blocking from my side!

Yes, this change does not really impact any existing APIs. It's adding a new delegate and will only be available for those who add this explicitly. Validated for backward compatibility.

**Pull request merged**